app/code/Magento/CheckoutAgreements/Block/Adminhtml/Agreement/Grid.php

```diff
@@ -8,21 +8,21 @@
 class Grid extends \Magento\Backend\Block\Widget\Grid\Extended
 {
     /**
-     * @var \Magento\CheckoutAgreements\Model\ResourceModel\Agreement\CollectionFactory
+     * @var \Magento\CheckoutAgreements\Model\ResourceModel\Agreement\Grid\CollectionFactory
      */
     protected $_collectionFactory;
 
     /**
      * @param \Magento\Backend\Block\Template\Context $context
      * @param \Magento\Backend\Helper\Data $backendHelper
-     * @param \Magento\CheckoutAgreements\Model\ResourceModel\Agreement\CollectionFactory $collectionFactory
+     * @param \Magento\CheckoutAgreements\Model\ResourceModel\Agreement\Grid\CollectionFactory $collectionFactory
      * @param array $data
      * @codeCoverageIgnore
      */
     public function __construct(
         \Magento\Backend\Block\Template\Context $context,
         \Magento\Backend\Helper\Data $backendHelper,
-        \Magento\CheckoutAgreements\Model\ResourceModel\Agreement\CollectionFactory $collectionFactory,
+        \Magento\CheckoutAgreements\Model\ResourceModel\Agreement\Grid\CollectionFactory $collectionFactory,
```

Comment:
Please, inject new dependencies according to our [Backward Compatible Development](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter) policy